### DISCUSSION TITLE
C++ namespaces for types

### DISCUSSION THREAD

**Discussion by vikke1234** [Vector35/binaryninja-api#6221] [Q&A] [answered]
Hi,

Are you able to somehow set namespaces for types?

It seems that whenever I try to give a struct for example a prefix of `foo::Bar` it allows me to name the struct that. But when I try to give that type to a variable it fails with "Error parsing specified type", once the `foo::` prefix is removed it accepts it.

**Comment by emesare** [comment, marked as answer]
Encapsulate the name with `

For example:
```
`foo::Bar`
```

<img width="532" alt="image" src="https://github.com/user-attachments/assets/1566963d-b424-4118-9cc8-8e6522898e6b">

**Reply by vikke1234** [reply]
Oh, awesome, thank you.

**Reply by emesare** [reply]
Here's the relevant page in the docs fwiw: https://docs.binary.ninja/guide/types/type.html?h=backtick#special-characters
